mlos_bench/mlos_bench/config/services/remote/azure/scripts/setup-rg/setup-rg.ps1

```diff
@@ -0,0 +1,115 @@
+
+param(
+    # Main control plane ARM template params
+    [Parameter(Mandatory=$True)]
+    [string] $controlPlaneArmParameters,
+    # Results DB ARM template params
+    [Parameter(Mandatory=$False)]
+    [string] $resultsDbArmParameters,
+    # Other params
+    [Parameter(Mandatory=$True)]
+    [string] $servicePrincipalName,
+    [Parameter(Mandatory=$True)]
+    [string] $resourceGroupName,
+    [Parameter(Mandatory=$True)]
+    [string] $certName,
+    [Parameter(Mandatory=$False)]
+    [int] $certExpirationYears = 1
+)
+
+# Provision resources into the resource group with ARM template
+Write-Output "Provisioning control plane resources..."
+$deploymentResults = az deployment group create `
+    --resource-group $resourceGroupName `
+    --template-file .\rg-template.json `
+    --parameters $controlPlaneArmParameters `
+    | ConvertFrom-JSON
+
+if (!$?) {
```

Comment:
Is that sufficient, or do you also need to check `$LASTEXITCODE`?

Reply:
I feel this is sufficient, since `$?` is always a boolean indicating success of the last command instead of explicit exit codes.

Reply:
I've run into trouble with that in the past, but it might have been due to behavior like this:

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.3#section-1

> Until PowerShell 7, wrapping a statement within parentheses `(...)`, subexpression syntax `$(...)`, or an array expression `@(...)` always reset `$?` to `True`. For example, `(Write-Error)` shows `$?` as `True`. This behavior changed in PowerShell 7, so that `$?` always reflects the actual success of the last command run in these expressions.

Reply:
Good to know! Will add a note about this in the README about using this script recommending Powershell 7

Reply:
Could add a `#Requires -Version 7` header to the top.

Reply:
Added it! Didn't realize that was a thing, thanks for letting me know!